week-1-ts-ecosystem/w01e01-typescript-basics/src/logic.ts

```diff
@@ -0,0 +1,9 @@
+import { IFunction, IUser } from "./types";
+
+export function hasGivenAge(requiredAge: number): IFunction {
+  return (user: IUser) => user.age >= requiredAge;
+}
+
+export function hasAddress(user: IUser): boolean {
+  return Boolean(user.address);
+}
```

Comment:
Generalnie sam TS i typowanie jest zrobione dobrze.
To co zwraca moją uwagę to przedrostek `I` przed typami. Generalnie nie jestem fanem tego typu rozwiązania zwłaszcza w `TypeScipt`. Nie za bardzo rozumiem czemu to rozróżnienie oznaczenia `Interface` miałoby służyć w sensie - czy faktycznie jest to tak istotne.

Dodatkowo przemyśl proszę nazewnictwo plików. 
Tam gdzie jest `logic.ts` widziałbym raczej to główne wpisanie danych i sprawdzenie ich.

`user.ts` to z kolei - model danych `User` - tak bym to widział.
to co tutaj masz w `logic.ts` to dla mnie pewnego rodzaju `validators` połączone z userem - może w ten sposób ?

Reply:
Dzięki za uwagi, będę bardziej zwracał uwagę na nazewnictwo